src/wms/src/main/java/org/geoserver/wms/legendgraphic/LegendUtils.java

```diff
@@ -641,7 +655,11 @@ public static BufferedImage renderLabel(
             if (labelBounds.getWidth() > req.getWidth()) {
                 FontMetrics fm = g.getFontMetrics();
                 int widthChars = req.getWidth() / fm.stringWidth("m");
-                WordUtils.wrap(label, widthChars, "\n", true);
+                int wrapLength = LegendUtils.getWrapLimit(req);
+                if (wrapLength != -1) {
+                    widthChars = Math.min(widthChars, wrapLength);
```

Comment:
Not 100% sure about this.
Wouldn't that result in always wrapping to the minimum widthChars.
Let's say you have a very long label and a small image and we set a wrapLength around 100.

that "min" will wrap a the widthChars whilst we wanted wrapping 100 chars instead.

Reply:
updated code